update 'update-codegen.sh' to make it works outside GOPATH

zwpaper opened the pull request:
Signed-off-by: Wei Zhang <kweizh@gmail.com>

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

/kind bug

#### What this PR does / why we need it:

https://github.com/kubernetes-sigs/scheduler-plugins/commit/b5ea0f1f188e37f770868c7a74a9ae91107dd4e1#diff-149dfe7bb29d1191dceae3a52915e750e64b7f87257a5fb309c29d3056e2a95d This commit previously deleted the `generate-group.sh`, but then can not gen clientset, informers and listers. 

this PR:
1. added the `generate-group.sh` back to generate clientset, informers, listers for scheduler-plugins
2. bump controller gen to `v0.11.1` to match https://github.com/kubernetes-sigs/scheduler-plugins/blob/master/hack/verify-crdgen.sh#L30
3. add `output-base` for codegen to make it works outside GOPATH

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

#### Special notes for your reviewer:

ref: https://github.com/kubernetes-sigs/kwok/pull/226

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note

```

Huang-Wei commented:
@zwpaper thanks for the PR, but I cannot merge as-is now.

There is a history for https://github.com/kubernetes-sigs/scheduler-plugins/commit/b5ea0f1f188e37f770868c7a74a9ae91107dd4e1#diff-149dfe7bb29d1191dceae3a52915e750e64b7f87257a5fb309c29d3056e2a95d. Given it's pretty common for scheduler plugins to manipulate CR objects, and more important, **_different_** CR objects. So it makes a lot of sense to not use generated clients (the traditional k/k way); instead, leverage controller-runtime and follow kubebuilder's annotation syntac to use general (dynamic) client to manipulate the CR objects. However, that migration (from static clients to controller-runtime's client) is half-way done due to lack of volunteers. We need to finish it up sometime so we don't need static clients anymore.

zwpaper commented:
Hi @Huang-Wei, got it! as for the volunteers part, where can I dig deeper into the controller-runtime topic? I would like to contribute to it, I used the operator from time to time, and maybe there is something I could help with.

Huang-Wei commented:
> Hi @Huang-Wei, got it! as for the volunteers part, where can I dig deeper into the controller-runtime topic? I would like to contribute to it, I used the operator from time to time, and maybe there is something I could help with.

I need to take a look at the codebase first.. to see where we were left at. And then come up with a plan to work on the migration. Stay tuned.

zwpaper commented:
Hi @Huang-Wei, I have updated the PR, and deleted the generated clientset related parts.

it is still necessary to do this fix because we still need `update-internal-group.sh` to generate deepcopy,defaults,conversion, and it should work outside GOPATH after this PR get merged.

also, I have updated the `controller-gen` to `0.11.1` to match https://github.com/kubernetes-sigs/scheduler-plugins/blob/master/hack/verify-crdgen.sh?rgh-link-date=2023-01-24T09%3A50%3A10Z#L30

zwpaper commented:
as for the controller runtime part, I will dig into the code base before your plan is ready😁

zwpaper commented:
happened to find this PR may fix https://github.com/kubernetes-sigs/scheduler-plugins/issues/445 as it failed due to not putting `scheduler-plugins` under GOPATH

Huang-Wei commented:
> I have updated the PR, and deleted the generated clientset related parts.

Thanks. The current state looks good.

/ok-to-test

Huang-Wei commented:
/lgtm
/approve
/release-note-none
